MetaMask/metamask-extension
# Docs/12367 Adding storybook essentials addons

Fixes: #12367

## Explanation
In an effort to improve the documentation of our components this PR updates storybook to the latest version and includes the storybook essential plugins. 

Using the `Button` component as an example it also includes storybook contribution guidelines and best practices.

#### TO DO
- [x] Update storybook to `^6.3.12`
- [x] Add [storybook essentials](https://storybook.js.org/docs/react/essentials/introduction) plugins
- [x] Add [storybook docs]() and MDX documentation
- [x] Add [a11y plugin](https://storybook.js.org/addons/@storybook/addon-a11y) for basic accessibility checking
- [x] Updated `Button` story and add documentation
- [x] Add contributor docs and storybook book best practices
- [x] Remove unnecessary updates and packages 
- [x] Removing dumpicate plugins that are included in `@storybook/addon-essentials`
- `@storybook/addon-backgrounds`
- `@storybook/addon-toolbars`

#### Manual testing steps:
- Go to [latest storybook build](https://364136-42009758-gh.circle-artifacts.com/0/storybook/index.html?path=/story/getting-started-introduction--page)
- View Getting Started and Documentation Guidelines (it's currently at the bottom of the nav but this will change when we get PR #12424 in)
- Go to `Button` story
- View Controls and check all interactions work
- View Accessibility tab to view basic accessibility tests 
- View Docs page to read documentation on component api and usage

## Images
<img width="1435" alt="Screen Shot 2021-10-25 at 6 56 00 PM" src="https://user-images.githubusercontent.com/8112138/138795592-58c4eb7c-6088-4112-979c-fdc43e623b5e.png">
<img width="1440" alt="Screen Shot 2021-10-25 at 6 56 08 PM" src="https://user-images.githubusercontent.com/8112138/138795599-27982cef-81f4-44b3-bf6b-b97a8693010f.png">
<img width="1439" alt="Screen Shot 2021-10-25 at 6 56 23 PM" src="https://user-images.githubusercontent.com/8112138/138795752-c729da32-a2ba-402e-8c38-ef7cb07d7deb.png">
<img width="1439" alt="Screen Shot 2021-10-25 at 6 57 04 PM" src="https://user-images.githubusercontent.com/8112138/138795767-e77a93a1-d291-445b-bd6d-feb825a8012f.png">
